### DISCUSSION TITLE
How can I custom syntax?

### DISCUSSION THREAD

**Discussion by By-Ha** [Milkdown/milkdown#1000] [Q&A] [answered]
As https://milkdown.dev/docs/plugin/example-iframe-plugin mentioned, it is easy to create a syntax like ::name[label]{attributes}.
But how can I create a syntax like `@username` and render it with react?

**Comment by Saul-Mirone** [comment, marked as answer]
Generally you'll need to use a remark plugin to create a node for your syntax, something similar to https://github.com/Milkdown/milkdown/blob/main/packages/plugin-emoji/src/__internal__/remark-twemoji.ts#L39

**Reply by By-Ha** [reply]
I have read this source code before. It did help.

**Reply by zgrybus** [reply]
@By-Ha Did you figure it out?

**Comment by By-Ha** [comment]
nope



发自我的iPhone


------------------ Original ------------------
From: Krzysztof Nofz ***@***.***&gt;
Date: Wed,Mar 6,2024 8:06 PM
To: Milkdown/milkdown ***@***.***&gt;
Cc: By_Ha ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [Milkdown/milkdown] How can I custom syntax? (Discussion #1000)





 
@By-Ha Did you figure it out?

**Reply by zgrybus** [reply]
@By-Ha 
If you want, I'm trying to develop this feature. There is my question about syntax -> https://github.com/orgs/Milkdown/discussions/1262

**Comment by zgrybus** [comment]
@By-Ha 
I've created a plugin that gives you possibility to mention someone https://github.com/zgrybus/milkdown-mentions-plugin :)
